specs/src/specs/shares.md

```diff
@@ -0,0 +1,90 @@
+- If there is insufficient transaction or PayForBlob transaction data to fill the share, the remaining bytes are filled with `0`.
+
+First share in a sequence:
+
+![fig: transaction share start](./figures/transaction_share_start.svg)
+
+where reserved bytes would be `38` as a binary big endian `uint32` (`[0b00000000, 0b00000000, 0b00000000, 0b00100110]`).
+
+Continuation share in a sequence:
+
+![fig: transaction share continuation](./figures/transaction_share_continuation.svg)
+
+where reserved bytes would be `80` as a binary big endian `uint32` (`[0b00000000, 0b00000000, 0b00000000, 0b01010000]`).
+
+## Padding
+
+Padding shares vary based on namespace but share a common structure:
+
+- The first [`NAMESPACE_SIZE`](./consensus.md#constants) of a share's raw data is the namespace of that share.
+- The next [`SHARE_INFO_BYTES`](./consensus.md#constants) bytes are for share information.
+  - The first 7 bits represent the share version in big endian form (initially, this will be `0000000` for version `0`);
+  - The last bit is a sequence start indicator. The indicator is always `1`.
+- The next [`SEQUENCE_BYTES`](./consensus.md#constants) contain a big endian `uint32` of value `0`.
+- The remaining [`SHARE_SIZE`](./consensus.md#constants)`-`[`NAMESPACE_SIZE`](./consensus.md#constants)`-`[`SHARE_INFO_BYTES`](./consensus.md#constants)`-`[`SEQUENCE_BYTES`](./consensus.md#constants) bytes are filled with `0`.
+
+### Namespace Padding Share
+
+A namespace padding share uses the namespace of the blob that precedes it in the data square so that the data square can retain the property that all shares are ordered by namespace.
+A namespace padding share acts as padding between blobs so that the subsequent blob may begin at an index that conforms to the [non-interactive default rules](../rationale/data_square_layout.md#non-interactive-default-rules). Clients can safely ignore the contents of these shares because they don't contain any significant data.
+
+### Reserved Padding Share
+
+Reserved padding shares use the [`RESERVED_PADDING_NAMESPACE`](./consensus.md#constants). Reserved padding shares are placed after the last reserved namespace share in the data square so that the first blob can start at an index that conforms to non-interactive default rules. Clients can safely ignore the contents of these shares because they don't contain any significant data.
+
+### Tail Padding Share
+
+Tail padding shares use the [`TAIL_PADDING_NAMESPACE`](./consensus.md#constants). Tail padding shares are placed after the last blob in the data square so that the number of shares in the data square is a perfect square. Clients can safely ignore the contents of these shares because they don't contain any significant data.
```

Comment:
I'd suggest using more precise wording (now that we are revising the specs), for example, 
Clients **MUST/SHOULD** ignore the contents of these shares because they don't contain any significant data.

Reply:
Good idea. Going to use MAY here based on https://www.ietf.org/rfc/rfc2119.txt

Reply:
Resolved via 307c6c07a9827e6a5e9c6236623e6570885adeb3

Reply:
Thanks for the changes in https://github.com/celestiaorg/celestia-app/commit/307c6c07a9827e6a5e9c6236623e6570885adeb3. Since these words are defined in RFC2119, we shall use capitalized format to emphasize that they should be interpreted according to that RFC.

Reply:
Resolved by 851467c234c7599c5bdadf801e58290c254cf2f0